src/devtools/views/Components/TreeContext.js

```diff
@@ -773,14 +773,34 @@ function recursivelySearchTree(
   regExp: RegExp,
   searchResults: Array<number>
 ): void {
-  const { children, displayName } = ((store.getElementByID(
+  const { children, displayName, hocDisplayNames } = ((store.getElementByID(
     elementID
   ): any): Element);
-  if (displayName !== null) {
-    if (regExp.test(displayName)) {
-      searchResults.push(elementID);
+  let resultWasPushed = false;
+
+  const pushResult = () => {
+    searchResults.push(elementID);
+    resultWasPushed = true;
+  };
+
+  if (
+    resultWasPushed === false &&
+    displayName != null &&
+    regExp.test(displayName) === true
+  ) {
+    pushResult();
+  }
+
+  if (
+    resultWasPushed === false &&
+    hocDisplayNames != null &&
+    hocDisplayNames.length > 0
+  ) {
+    if (hocDisplayNames.some(name => regExp.test(name) === true) === true) {
+      pushResult();
```

Comment:
Seems like we could simplify this a bit and remove the overhead of an extra function call by using an if/else. What do you think?

```js
function recursivelySearchTree(
  store: Store,
  elementID: number,
  regExp: RegExp,
  searchResults: Array<number>
): void {
  const { children, displayName, hocDisplayNames } = ((store.getElementByID(
    elementID
  ): any): Element);

  if (
    displayName != null &&
    regExp.test(displayName) === true
  ) {
    searchResults.push(elementID);
  } else if (
    hocDisplayNames != null &&
    hocDisplayNames.length > 0 &&
    hocDisplayNames.some(name => regExp.test(name))
  ) {
    searchResults.push(elementID);
  }

  children.forEach(childID =>
    recursivelySearchTree(store, childID, regExp, searchResults)
  );
}
```

Reply:
Great!